docs/STYLES.md

```diff
@@ -0,0 +1,88 @@
+
+Please check [a list of current styling issues](https://github.com/organicmaps/organicmaps/issues?q=is%3Aopen+is%3Aissue+label%3AStyles)
+and ["icons wanted" issues](https://github.com/organicmaps/organicmaps/issues?q=is%3Aopen+is%3Aissue+label%3AIcons+label%3A%22Good+first+issue%22).
+
+## Requirements
+
+To work with styles first [clone the OM repository](INSTALL.md#getting-sources).
+
+To run some scripts additional libraries are needed:
+```
+optipng
+protobuf 3.17.3 (3.18.0 doesn't work!)
+```
+
+## Files
+
+Map styles are defined in text files located in `data/styles/clear/include/`:
+* forests, rivers, buildings, etc. [`Basemap.mapcss`](../data/styles/clear/include/Basemap.mapcss)
+* text labels for above features [`Basemap_label.mapcss`](../data/styles/clear/include/Basemap_label.mapcss)
+* roads, bridges, foot and bicycle paths, etc. [`Roads.mapcss`](../data/styles/clear/include/Roads.mapcss)
+* [`Roads_label.mapcss`](../data/styles/clear/include/Roads_label.mapcss)
+* icons for POIs and other features [`Icons.mapcss`](../data/styles/clear/include/Icons.mapcss)
+* city-specific subway networks [`Subways.mapcss`](../data/styles/clear/include/Subways.mapcss)
+* light (default) theme colors: [`style-clear/colors.mapcss`](../data/styles/clear/style-clear/colors.mapcss)
+* dark/night theme colors: [`style-night/colors.mapcss`](../data/styles/clear/style-night/colors.mapcss)
+
+There is a separate set of these style files for the navigation mode in `data/styles/vehicle/`.
+
+Icons are stored in [`data/styles/clear/style-clear/symbols/`](../data/styles/clear/style-clear/symbols/) and their dark/night counterparts are in [`data/styles/clear/style-night/symbols/`](../data/styles/clear/style-night/symbols/).
+
+## How to add a new icon
+
+1. add an svg icon to `data/styles/clear/style-clear/symbols/` (and to `style-night` too)
+preferably look for icons in [collections OM uses already](../data/copyright.html#icons)
+2. add icon rendering/visibility rules into `data/styles/clear/include/Icons.mapcss` and to "navigation style" `data/styles/vehicle/include/Icons.mapcss`
+3. run `tools/unix/generate_symbols.sh` to add new icons into skin files
+4. run `tools/unix/generate_drules.sh` to generate drawing rules for the new icons
+
+## How to add a new map feature / POI type
+
+1. add it into `data/mapcss-mapping.csv` (or better replace existing `deprecated` line) to make OM import it from OSM
+2. if necessary merge similar tags in via `data/replaced_tags.txt`
+3. add a new icon (see [above](#how-to-add-a-new-icon))
+4. if a new POI should be OSM-addable/editable then add it to `data/editor.config`
+5. add new type translation into `data/types_strings.txt`
+6. add search keywords into `data/categories.txt`
+7. run `tools/unix/generate_localizations.sh` to validate and distribute translations into iOS and Android
+8. add new or fix current classifier tests at `/generator/generator_tests/osm_type_tests.cpp` if you can
```

Comment:
I'd leave "if you can" because if we make this step mandatory only cpp programmers will be able to contribute to the styles..

I'm happy to add more details if @vng will provide them. Probably it'll be better to add explanation comments in the beginning of the tests file and/or comment 1-2 of existing test cases well so that its easier for new contributors to adapt them into new test cases.

Reply:
Let's mark this number 8 as optional .. Ok, will add more comments in osm_type_tests.cpp module someday

Reply:
Ok, great!
I think "if you can" suggests that its otptional already.